app/v1/services/EnrolmentsAuthService.scala

```diff
@@ -38,18 +40,55 @@ class EnrolmentsAuthService @Inject()(val connector: AuthConnector) {
     override def authConnector: AuthConnector = connector
   }
 
-  def authorised(predicate: Predicate)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[AuthOutcome] = {
-    authFunction.authorised(predicate).retrieve(affinityGroup and allEnrolments) {
-      case Some(Individual) ~ enrolments => createUserDetailsWithLogging(affinityGroup = "Individual", enrolments)
-      case Some(Organisation) ~ enrolments => createUserDetailsWithLogging(affinityGroup = "Organisation", enrolments)
-      case Some(Agent) ~ enrolments => createUserDetailsWithLogging(affinityGroup = "Agent", enrolments)
-      case _ =>
-        Logger.warn(s"[AuthorisationService] [authoriseAsClient] Authorisation failed due to unsupported affinity group.")
-        Future.successful(Left(LegacyUnauthorisedError))
-    } recoverWith unauthorisedError
+  def authorised(predicate: Predicate, nrsRequired: Boolean = false)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[AuthOutcome] = {
```

Comment:
I feel like this might be better as two functions (one for nrs, and one without) rather than a single function with both. Combining the two together makes this method a bit hard to read.

Reply:
If we create a separate function we need to modify the **AuthorisedController.authorisedAction**. or we have to create another function here also.